src/develop/imageop.c

```diff
@@ -2341,6 +2341,22 @@ void add_remove_mask_indicator(dt_iop_module_t *module, gboolean add)
                      G_CALLBACK(_display_mask_indicator_callback), module);
     gtk_widget_set_sensitive(module->mask_indicator, !raster && module->enabled);
     gtk_box_pack_end(GTK_BOX(module->header), module->mask_indicator, FALSE, FALSE, 0);
+
+    // in dynamic modes, we need to put the mask indicator after the drawing area
+    GList *children = gtk_container_get_children(GTK_CONTAINER(module->header));
+    GList *child = g_list_last(children);
+
+    for(child = g_list_last(children); child && GTK_IS_BUTTON(child->data); child = g_list_previous(child));
+
+    if(GTK_IS_DRAWING_AREA(child->data))
+      {
```

Comment:
indentation is not dt style.

Can you also explain more what this does? Maybe a screenshot ?

Reply:
It should basically fix the bug listed in #9073. The module-header mask indicator is not currently appearing for some of the modes where the right-hand-side module-header icons are auto-hidden.

Reply:
@elstoc : Yes, but my question is more how was this fixed, the changelog  says "by placing the mask indicator to the right of the drawing area for such modes" ... And I'm wondering where exactly the mask is now drawn as this is not clear to me, and difficult to see from just reading code.

Reply:
There's some explanation [here](https://github.com/darktable-org/darktable/issues/9073#issuecomment-849100989)